### PR TITLE
reland interface nested into subroutine causes symbol undefined, fix …

### PR DESCRIPTION
**Pull request by pwisniewskimobica**
…#1013 #1179

add test and remove init_use_tree

add empty line in test file

cosmetic fixes

add empty line at the end of test file

add spaces

correct the coding style

add declaration

### CODE REVIEW

**Review by pawosm-arm — APPROVED** [review]

**Comment by pawosm-arm** [comment]
@kiranchandramohan @alokkrsharma and @bryanpkc can I kindly ask you to look at this again?

File: tools/flang1/flang1exe/module.c [hunk 1]
```diff
@@ -371,7 +372,11 @@ apply_use_stmts(void)
   save_lineno = gbl.lineno;
 
   if (!gbl.currmod && gbl.internal <= 1) {
-    init_use_tree();
+    for (m_id = FIRST_USER_MODULE; m_id < usedb.avl; m_id++) {
+      if (usedb.base[m_id].scope > sem.scope_level) {
```
Comment (kiranchandramohan) [inline tools/flang1/flang1exe/module.c:376]: Nit: Can you add a comment here explaining why removing from the tree is the right action here.

**Review by kiranchandramohan — APPROVED** [review]
LG. Minor nit.

**Comment by pwisniewskimobica** [comment]
Removing module from the tree is needed here to not taking into processing the module beyond the scope.
Before we did init_use_tree which was removing every module not only the needed one.

**Comment by kiranchandramohan** [comment]
Since this was approved previously, we can land this again without new approvals.

**Pull request merged**